fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2771,6 +2781,10 @@ public Range getUnbuiltRange() {
      * Add a read conflict key for all records.
      */
     private void addRecordsReadConflict() {
+        if (recordsReadConflict) {
+            return;
+        }
+        recordsReadConflict = true;
```

Comment:
This method is only called if one wants to add a conflict range over all records, so it should essentially never be called. I'm surprised it's showing up in your profiling. Maybe it doesn't hurt to also optimize it in this way.